modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpRequestHandler.java

```diff
@@ -51,9 +54,12 @@ class Netty4HttpRequestHandler extends SimpleChannelInboundHandler<HttpPipelined
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, HttpPipelinedRequest httpRequest) {
         final Netty4HttpChannel channel = ctx.channel().attr(Netty4HttpServerTransport.HTTP_CHANNEL_KEY).get();
+        final RestResponse earlyResponse = ctx.channel().attr(Netty4HttpServerTransport.EARLY_RESPONSE).get();
```

Comment:
Looks like we ran into an issue where we weren't able to leave the pipeline early if the request should be rejected - I think we should try to avoid this.  I'll try to get more details.

Reply:
There are certain events that occur now, that may not occur if the pipeline is exited early ([example](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java#L425)). It may also be important that all requests hit the [Netty4HttpPipeliningHandler](https://github.com/opensearch-project/OpenSearch/blob/main/modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpPipeliningHandler.java#L48-L50) as well. I'm not sure if its possible to short-circuit completely, but that would be another good improvement if possible.

Reply:
This has been resolved.